spec/ndt7-protocol.md

```diff
@@ -192,7 +192,158 @@ implementations will likely round bigger (or smaller) numbers to the nearest
 make sure that it does not mit values that a `int53` cannot represent. The
 proper action to take in this case is currently unspecified.
 
-# Reference implementation
+## Server discovery
+
+This section explains how a client can get the FQDN of a ndt7-enabled
+M-Lab server for the purpose of running tests. Of course, this
+section only applies to clients using M-Lab infrastructure. Clients
+using other server infrastructure MUST refer to the documentation
+provided by such infrastructure instead.
+
+Clients:
+
+1. SHOULD use the [locate.measurementlab.net](
+https://locate.measurementlab.net/) server-discovery web API;
+
+2. MUST query for the `ndt7` locate.measurementlab.net tool (see below for a
+description of how a real request would look like);
+
+3. MUST set `User-Agent` to identify themselves;
+
+4. MUST correctly handle `3xx` redirects, interpret `200` as success, and
+any other status code as failure;
+
+5. MUST handle gracefully the case where M-Lab is out of capacity, which
+is identifiable by locate.measurementlab.net returning `200` along with an
+empty response.
+
+The following example shows a request to locate.measurementlab.net originating
+from a well-behaved ndt7 client:
+
+```
+* Connected to locate.measurementlab.net (216.58.205.84) port 443 (#0)
+> GET /ndt7 HTTP/2
```

Comment:
Note to self: use `HTTP/1.1`: cURL actually represents HTTP/2 in a way that looks like HTTP/1.1, but HTTP/2 is different and has pseudo-headers. So I don't want to misrepresent HTTP/2 here.